lib/datadog/opentelemetry/sdk/span_processor.rb

```diff
@@ -31,6 +32,15 @@ def on_start(span, parent_context)
         #
         # @param [Span] span the {Span} that just ended.
         def on_finish(span)
+          unless span.events.nil?
+            span.datadog_span.span_events = span.events.map do |event|
```

Comment:
The Opentelemetry SDK does not return the Event added via `Span.add_event` method. Overriding Span.add_event and accessing the last event in the list can introduce race conditions: https://github.com/open-telemetry/opentelemetry-ruby/blob/406cabe54f8fb4937d0f6b73fe830a51c04fdb92/sdk/lib/opentelemetry/sdk/trace/span.rb#L173.

The simplest way to convert OpenTelemetry Span Events to Datadog Span Events is on span finish.